### PR TITLE
Handling wrong authentication.

### PR DESCRIPTION
**Pull request by roccomuso**
Done: emit an error on console if credentials are wrong, (both for pub and sub).

Todo: gotta be handled the 'No Authorizations' case.
I did some attempts on sub.js, editing the client.on('connect') callback this way:

      client.on('connect', function() {
        client.subscribe(args.topic, { qos: args.qos }, function(err, granted){
          if (err){
            console.warn('Error: ', err);
            client.end();
          }else console.log('Granted: ', granted);
        });

and this is my output on a topic that i'm not authorized to subscribe to:

    Granted: [ { topic: 'priv8', qos: 128 } ]

Weird, isn't it? The broker i'm using is Mosca. I'm implementing on Mosca the following functions: authenticate, authorizePublish, authorizeSubscribe.

### CODE REVIEW

File: bin/sub.js [hunk 1]
```diff
@@ -99,6 +99,14 @@ function start(args) {
       console.log(payload.toString())
     }
   });
+  
+  client.on('error', function(err){
+    // Handle 'No Authentication'
+    console.warn(err);
+    client.end();
+  });
+  // TODO: alert impossibility to subscribe to a topic, 'No authorization'.
```
Comment (mcollina) [inline bin/sub.js]: While you are at it, we should fix this TODO as well :).

**Comment by roccomuso** [comment]
Is 'no authorization' error handled somewhere else? Do we need to catch any 'particular' mqtt packet?

**Comment by mcollina** [comment]
If a subscription fails, you would find something different from 0,1,2 as the qos in the first argument of the `subscribe` callback: https://github.com/mqttjs/MQTT.js/blob/master/lib/client.js#L801-L805.

**Comment by roccomuso** [comment]
Subscription 'no authorization' alert handled: ab40be81c01cf0e85f09ad17efc48cce22fea67b

Any tips for the publish?

**Comment by mcollina** [comment]
Not really, an unathorized publish will disconnect the client, there is no feedback from the broker.

**Comment by mcollina** [comment]
Is this done?

**Comment by roccomuso** [comment]
Not yet. How do we handle this, authorized or not, a client will be either disconnected, am i right?

**Comment by mcollina** [comment]
Yes. But not really. in one case you'd get a 'bad' disconnect triggered from the server, on the other it will actually disconnect cleanly.
It's probably impossible to tell with QoS 0, as it disconnects straight away. But it should be possible to do something for QoS 1 & 2.

**Comment by mcollina** [comment]
Merged with #345

**Pull request closed without merging**